### DISCUSSION TITLE
UTF-8 implementation on LS

### DISCUSSION THREAD

**Discussion by SpeedyCharly** [AzuraCast/AzuraCast#4925] [Q&A] [answered]
A while back I asked about why UTF-8 doesn't recognize standard French characters in the metadata as seen here; 
-----------------------------------
2021/12/25 09:24:18 [camomile:3] Failed to convert "Fetes - Yvan- V\233ronique  en ondes" from UTF-8 to UTF-8 (CamomileLibrary__CharEncoding.Configure(Config).Malformed_code)!
2021/12/25 09:24:18 [input_streamer:3] New metadata chunk ? -- Fetes - Yvan- V?ronique  en ondes.
-----------------------------------
At the same time we poll the metadata from our Shoutcast stream [/7.html] to feed our chatroom bot which shows up properly:
![UTF](https://user-images.githubusercontent.com/66107794/147392524-c2fcb8c5-5021-4f4d-bef2-ce4c312eff6b.jpg)

Didn't really get a response the first time as it was embedded into another topic...

So, the question is; what does Shoutcast do that LS doesn't?

Could the problem reside with the Unicode library used?
[Camomile is a Unicode library for objective caml (a. k. a. OCaml or O'Caml) language. Camomile provides Unicode character type, UTF-8, UTF-16, UTF-32 strings, conversion to/from about 200 encodings, collation and locale-sensitive case mappings, and more. The library is currently designed for Unicode Standard 3.2.]

**Comment by SpeedyCharly** [comment, marked as answer]
Guess someone must have done something on this; havent seen metadata errors in 2-3 days now... Thanks to whomever...

**Reply by BusterNeece** [reply]
On Liquidsoap, we did switch from using cURL commands to directly calling liquidsoap's internal `http` client functions, which may have improved that situation quite a lot.
